examples/chef/common/clusters/oven-mode/chef-oven-mode.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "chef-oven-mode.h"
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/endpoint-config-api.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#ifdef MATTER_DM_PLUGIN_OVEN_MODE_SERVER
+
+using namespace chip;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ModeBase;
+using chip::Protocols::InteractionModel::Status;
+template <typename T>
+using List              = chip::app::DataModel::List<T>;
+using ModeTagStructType = chip::app::Clusters::detail::Structs::ModeTagStruct::Type;
+
+namespace ChefOvenMode {
+
+constexpr size_t kOvenModeTableSize = MATTER_DM_OVEN_MODE_CLUSTER_SERVER_ENDPOINT_COUNT;
+static_assert(kOvenModeTableSize <= kEmberInvalidEndpointIndex, "OvenMode table size error");
+
+std::unique_ptr<OvenMode::ChefDelegate> gDelegateTable[kOvenModeTableSize];
+std::unique_ptr<ModeBase::Instance> gInstanceTable[kOvenModeTableSize];
+
+/**
+ * Initializes OvenMode cluster for the app (all endpoints).
+ */
+void InitChefOvenModeCluster()
+{
+    const uint16_t endpointCount = emberAfEndpointCount();
+
+    for (uint16_t endpointIndex = 0; endpointIndex < endpointCount; endpointIndex++)
+    {
+        EndpointId endpointId = emberAfEndpointFromIndex(endpointIndex);
+        if (endpointId == kInvalidEndpointId)
+        {
+            continue;
+        }
+
+        // Check if endpoint has OvenMode cluster enabled
+        uint16_t epIndex =
+            emberAfGetClusterServerEndpointIndex(endpointId, OvenMode::Id, MATTER_DM_OVEN_MODE_CLUSTER_SERVER_ENDPOINT_COUNT);
+        if (epIndex >= kOvenModeTableSize)
+            continue;
+
+        gDelegateTable[epIndex] = std::make_unique<OvenMode::ChefDelegate>();
+        gDelegateTable[epIndex]->Init();
+
+        uint32_t featureMap = 0;
+        VerifyOrDieWithMsg(OvenMode::Attributes::FeatureMap::Get(endpointId, &featureMap) == Status::Success, DeviceLayer,
```

Comment:
It's too bad that we rely on globally reading the featuremap set via ZAP for this... I don't think there's a way to avoid that right now, but it's something to really remember relative to integration "institutional knowledge" that some of the config comes from ZAP files.

Reply:
Working in #38121 to expose this in some clearer way ... although getting the features from zap seems what we currently need to do. Generally Accessors are very tied to zap as they rely on memcpy of data. We could update them to use DataModel, however DataModel encode/decode has high overhead due to AAI requiring AttributeDataIB encoding (which is even more complex than raw TLV encoding)